src/main/java/com/hiqes/andele/RequestOwnerAndroidXAppCompatActivity.java

```diff
@@ -96,11 +96,11 @@ public boolean shouldShowRequestPermissionRationale(String permission) {
 
     @Override
     public boolean isSameOwner(RequestOwner otherOwner) {
-        boolean                         ret = false;
-        RequestOwnerAppCompatActivity   otherOwnerAct = null;
+        boolean                               ret           = false;
+        RequestOwnerAndroidXAppCompatActivity otherOwnerAct = null;
 
         try {
-            otherOwnerAct = (RequestOwnerAppCompatActivity)otherOwner;
```

Comment:
This likely needs to cascade between casts to `RequestOwnerAppCompatActivity` and `RequestOwnerAndroidXAppCompatActivity`.  Both cases need to be covered, not just AndroidX.

Reply:
Actually, I now see that you refactored this class completely.  We will need 2 different classes: one for the old app compat and one for Android X app compat.

Reply:
Changes done as discussed.